File: missing-ranges/missing-ranges.js
/**
 * @param {number[]} nums
 * @param {number} lower
 * @param {number} upper
 * @return {string[]}
 */
var findMissingRanges = function(nums, lower, upper) {
    
    
    let res = []; 
    let data = [lower-1, ...nums, upper+1];
    
    
    for(let i=0; i < data.length; i++){
        
        let diff = data[i+1] - data[i];
        
        if(diff === 2){
            res.push(`${data[i]+1}`)
        }else if (diff > 2){
            res.push(`${data[i]+1}->${data[i+1]-1}`)
        }
    }
    

    return res;
};

File: missing-ranges/missing-ranges.py
class Solution:
    def findMissingRanges(self, nums: List[int], lower: int, upper: int) -> List[str]:
        data = [lower-1] + nums + [upper+1]
        print(data)
        
        ans = []
        
        for i in range(1, len(data)):
            diff = data[i] - data[i-1]
            
            if diff == 2:
                ans.append(str(data[i]-1))
            elif diff > 2:
                val = str(data[i-1]+1) + "->"+ str(data[i]-1)
                ans.append(val)
        
        return ans